doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+* `tabRow.background`: Control the color of the background of the tab row items.
+  See below for accepted colors.
+* `pane.borderColor`: Control the color of the border used to separate panes.
+  This is the color of the inactive border between panes.
+* `pane.activeBorderColor`: Control the color of the border of the active pane
+* `pane.borderWidth`: Control the width of the borders used to separate panes.
+* `window.applicationTheme`: If set, will set the XAML `RequestedTheme`
+  property. This can be one of `light`, `dark` or `system`. This controls how
+  XAML fundamentally styles UI elements. If not provided, will use the default
+  value "system", which will use whatever the system's default theme is.
+
+#### Theme Colors
+
+For properties like `tab.background` and `tabRow.background`, these colors can
+be one of:
+* an `#rrggbb`, `#aarrggbb` color. (Alpha is ignored for `tabRow.background`)
+* `accent` for the accent color
+* `terminalBackground` to use the default background color of the active
+  terminal instance.
+* `terminalForeground` to use the default foreground color of the active
+  terminal instance.
+* `key:SomeXamlKey` to try and look `SomeXamlKey` up from our resources as a
+  `Color`, and use that color for the value.
+    - TODO DISCUSSION: Does anyone want this?
+    - is `accent` just `key:SystemAccentColor`? If it is, is it a reasonable
+      alias that we'd want to provide anyways?
+    - TODO DISCUSSION: PR[#5280] suggested `{ "key": "SomeResourceKey" }` for
+      string resources, should we use that format for colors like this as well?
+
+This will enable users to not only provide custom colors, but also use the
+dynamic color of the active terminal instance as well.
+
+Using `terminalBackground` with multiple concurrent panes with different
+backgrounds could certainly lead to some odd behavior. The intention of the
+setting is to provide a way for the tab/titlebar to "seamlessly" connect to the
+terminal content. However, two panes side-by-side could have different
+background colors, which might have an unexpected appearance. Since the user
+must have opted in to this behavior, they'll need to decide personally if
+that's something that bothers them aesthetically. It's entirely possible that a
+user doesn't use panes, and this wouldn't even be a problem for them.
+
+<!-- We could maybe mitigate this by providing the user a way of specifying the
+`tab.background` color as having both a "single pane" and "multiple pane" mode,
+though I'm not sure I'm in love with this:
+
+```json
+    "tab.background": {"single": "terminalBackground", "multiple": null},
+    "tab.background": {"single": "terminalBackground", "multiple": "#ff0000"},
+    "tab.background": [ "terminalBackground", null ]
+```
+
+ Also shown is an array based implementation, as an option. Overall I'm not
+ happy with this, so I think it shouldn't be in the final draft of the spec, but
+ I'm leaving it for now as an option.
+-->

```

Comment:
_This alone_ could get its own spec hahaha. I think following the active terminal instance is good for now. I'm thinking of some weird behaviors we could do that I think just isn't worth dealing with in v1:
| Scenario | Behavior |
|--|--|
| vertical stack of panes | (1) inherit top color |
| horizontal stack of panes | (1) blend colors touching titlebar, (2) blend at pane separator |